core/src/main/java/org/apache/calcite/util/Sarg.java

```diff
@@ -149,4 +150,33 @@ public boolean isComplementedPoints() {
         && rangeSet.complement().asRanges().stream()
             .allMatch(RangeSets::isPoint);
   }
+
+  /** Returns a measure of the complexity of this expression.
+   *
+   * <p>It is basically the number of values that need to be checked against
+   * (including NULL).
+   *
+   * <p>Examples:
+   * <ul>
+   *   <li>{@code x = 1}, {@code x <> 1}, {@code x > 1} have complexity 1
+   *   <li>{@code x > 1 or x is null} has complexity 2
+   *   <li>{@code x in (2, 4, 6) or x > 20} has complexity 4
+   *   <li>{@code x between 3 and 8 or x between 10 and 20} has complexity 2
+   * </ul>
+   */
+  public int complexity() {
+    int complexity;
+    if (rangeSet.asRanges().size() == 2
+        && rangeSet.complement().asRanges().size() == 1
+        && RangeSets.isPoint(
+            Iterables.getOnlyElement(rangeSet.complement().asRanges()))) {
+      complexity = 1;
+    } else {
+      complexity = rangeSet.asRanges().size();
```

Comment:
Should this be `int complexity = rangeSet.asRanges().size()`, and then `if (...) { complexity = 1;}`?

Reply:
It's a matter of taste. I prefer not to re-assign variables.

Reply:
Then please avoid re-computing the same things again and again.

It is hard to review and analyze how these `.asRanges().size()` differ.